packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -107,102 +118,108 @@ class UserSessionManager implements IUserSessionManager {
     };
     // If auto session tracking is enabled start the session tracking
     if (state.session.sessionInfo.value.autoTrack) {
-      this.startAutoTracking();
+      this.startOrRenewAutoTracking();
+    }
+  }
+
+  /**
+   * Handles error
+   * @param error The error object
+   */
+  onError(error: Error | unknown): void {
+    if (this.errorHandler) {
+      this.errorHandler.onError(error, 'UserSessionManager');
+    } else {
+      throw error;
+    }
+  }
+
+  /**
+   * A function to sync object type values in storage
+   * @param key
+   * @param value
+   */
+  syncObjectInStorage(key: string, value: Nullable<ApiObject> | undefined) {
```

Comment:
You can combine this and `syncIdsInStorage`. The object and valid value checks can be applied to both type of values.

You can rename the method to `syncValueToStorage`.

Reply:
Addressed